ext/ffi/ffi.c

```diff
@@ -2919,7 +2929,7 @@ ZEND_METHOD(FFI, cdef) /* {{{ */
 
 	if (code && ZSTR_LEN(code)) {
 		/* Parse C definitions */
-		FFI_G(default_type_attr) = ZEND_FFI_ATTR_STORED;
+		FFI_G(default_type_attr) = 0;
```

Comment:
I afraid this was a design decision - cdef should relive all CData objects. Otherwise almost any type has to be remembered. The problem is not limited to return type, but to any declared type.

```php
<?php
$cdef = FFI::cdef("typedef int T[5];");
$a = $cdef->new("T");
var_dump($a);
unset($cdef);
var_dump($a);
```

It would be great to find a common efficient solution. Fixing this only for return type doesn't make a lot of sense.

I would agree to merge this fix only for return type without this ``s/ZEND_FFI_ATTR_STORED/0/`` hunk.

Reply:
Oh I see, thanks for checking. I'll see later this week if I can come up with something.

Reply:
Thanks for looking into this :)